go/updater/trainingJobUpdater.go

```diff
@@ -0,0 +1,81 @@
+package updater
+
+import (
+	padv1 "github.com/PaddlePaddle/cloud/go/apis/paddlepaddle/v1"
+	trainingJobClient "github.com/PaddlePaddle/cloud/go/client/clientset/versioned"
+	log "github.com/golang/glog"
+	"k8s.io/client-go/kubernetes"
+	"time"
+)
+
+const (
+	retryTimes            = 5
+	convertedTimerTicker  = 10 * time.Second
+	confirmResourceTicker = 5 * time.Second
+)
+
+type trainingJobEventType string
+
+const (
+	trainingJobEventDelete trainingJobEventType = "Delete"
+	trainingJobEventModify trainingJobEventType = "Modify"
+)
+
+type trainingJobEvent struct {
+	// pet is the TrainingJobEventType of TrainingJob
+	pet trainingJobEventType
+	// The job transfer the information fo job
+	job *padv1.TrainingJob
+}
+
+// TrainingJobUpdater is to manager a specific TrainingJob
+type TrainingJobUpdater struct {
+	// Job is the job the TrainingJob manager.
+	job *padv1.TrainingJob
+
+	// kubeCli is standard kubernetes client.
+	kubeCli kubernetes.Interface
+
+	// TrainingJobClient is the client of TrainingJob.
+	trainingJobClient trainingJobClient.Interface
```

Comment:
What is the difference between `kubeCli` and `trainingJobClient`? Shall we use pointer here?

Reply:
`kubeCli` only have the functions of k8s native resources. We need to use the `trainingJobClient` which @m3ngyang committed in  [#624](https://github.com/PaddlePaddle/cloud/pull/624).

Reply:
> Shall we use pointer here?

Interface is the better choice. In golang, interface is a spec that defines the behavior of a given object and Kubernetes also adopts Interface-based programming.

Reply:
@m3ngyang I see, thanks!